gateway/src/bt/uplink.c

```diff
@@ -93,17 +119,36 @@ void gateway_uplink_start(struct bt_conn *conn)
         return;
     }
 
-    struct bt_gatt_read_params *read_params = &node->read_params;
-    memset(read_params, 0, sizeof(*read_params));
-
-    read_params->func = tf_uplink_read_cb;
-    read_params->handle_count = 1;
-    read_params->single.handle = node->attr_handles[GOLIOTH_GATT_ATTR_UPLINK];
-    int err = bt_gatt_read(conn, read_params);
-    if (err)
+    if (node->attr_handles[GOLIOTH_GATT_ATTR_UPLINK].ccc)
     {
-        LOG_ERR("BT read request failed: %d", err);
-        bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+        struct bt_gatt_subscribe_params *subscribe_params = &node->subscribe_params;
+        memset(subscribe_params, 0, sizeof(*subscribe_params));
+
+        subscribe_params->notify = tf_uplink_indicate_cb;
+        subscribe_params->value = BT_GATT_CCC_INDICATE;
+        subscribe_params->value_handle = node->attr_handles[GOLIOTH_GATT_ATTR_UPLINK].value;
+        subscribe_params->ccc_handle = node->attr_handles[GOLIOTH_GATT_ATTR_UPLINK].ccc;
+        int err = bt_gatt_subscribe(conn, subscribe_params);
+        if (err)
+        {
+            LOG_ERR("BT subscribe request failed: %d", err);
+            bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```

Comment:
For consistency (can we enforce?):
```suggestion
            (void) bt_conn_disconnect(conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
```